openapi/find-my-trn-api.yml

```diff
@@ -0,0 +1,291 @@
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: teacher-identity
+servers:
+  - url: 'http://localhost:8080'
+paths:
+  /trn-email-match:
```

Comment:
Could this not be part of `/trn-details-match:`?

Reply:
Do we need the "match" at all? and could the endpoint be more descriptive i.e. `get-trn`?

Reply:
Could maybe have a single operation - the one at line 31 - and have it support passing _only_ an email address 🤔

Reply:
it could Dan, personally I prefer more defined interfaces, that do what they say on the tin (as much as possible). ATM the happy path in findmytrn is simply email, you would only pass the (many) optional params if there was no match on email. Under the covers they could use the same / similar code (leave that to @gunndabad :) ). We decided the current thinking is Register would have a separate find and create endpoints.

Reply:
it could Dan, personally I prefer more defined interfaces, that do what they say on the tin (as much as possible). ATM the happy path in findmytrn is simply email, you would only pass the (many) optional params if there was no match on email. Under the covers they could use the same / similar code (leave that to @gunndabad :) ). We decided the current thinking is Register would have a separate find and create endpoints.

Reply:
Makes sense @aje54